addons/purchase/models/purchase_order.py

```diff
@@ -147,11 +147,30 @@ def _compute_invoice(self):
     currency_rate = fields.Float("Currency Rate", compute='_compute_currency_rate', compute_sudo=True, store=True, readonly=True, help='Ratio between the purchase order currency and the company currency')
 
     mail_reminder_confirmed = fields.Boolean("Reminder Confirmed", default=False, readonly=True, copy=False, help="True if the reminder email is confirmed by the vendor.")
-    mail_reception_confirmed = fields.Boolean("Reception Confirmed", default=False, readonly=True, copy=False, help="True if PO reception is confirmed by the vendor.")
+    is_rfq_acknowledged = fields.Boolean("RFQ Acknowledged", default=False, readonly=True, copy=False, help="True if RFQ is confirmed/declined by the vendor.")
+    mail_reception_declined = fields.Boolean("Reception Declined", default=False, readonly=True, copy=False, help="True if PO reception is declined by the vendor.")
```

Comment:
Not sure to understand to which part of the spec those fields are related and why `mail_reception_confirmed` was not already enough ?
Also, `default` value for boolean is already `False`, no need to write it

Reply:
Hello @svs-odoo , according to the previous disccussion with Thomas(thd) we were continuously updating the chatter as user should have the flexibility to change the decision. Though we can check one each if once the user accept then log for that should only be once and same goes for decline.
Seeking confirmation for functional changes @thomasdeleval.

Reply:
For example, if we take the current state of `decline_reception_mail`:
```python
def decline_reception_mail(self):
    for order in self:
        if order.state in ['purchase', 'done'] and not order.mail_reception_declined:
            order.mail_reception_declined = True
            order.sudo().activity_schedule(
                'mail.mail_activity_data_todo',
                note=_('The vendor asked to decline this confirmed RfQ, if you agree on that, cancel this PO'))
        elif order.state in ['purchase', 'done']:
            order.sudo().message_post(body=_("The order receipt has been declined by %s.", order.partner_id.name))
        elif order.state  == 'sent':
            order.is_rfq_acknowledged = True
            order.sudo().message_post(body=_("The RFQ has been declined by %s.", order.partner_id.name))
```
* If the RFQ is `sent`, the RFQ will be marked as acknowledged and a message will be poster every time the user click on the decline button which seem wrong.
* If it purchase or done state:
  * If it's the first time the PO is declined, it will marked as declined and a message will be displayed;
  * If it's already marked as declined, another (but not the same) message will be displayed. Why do you have two messages for decline ?
And in both case, a message will be displayed every time the user click on the button which seems wrong.

Also, for the moment I still don't see the upside of having two fields (`is_rfq_acknowledged` and `mail_reception_declined`) instead of only one (previously, `mail_reception_confirmed`)

Reply:
Hello @svs-odoo, as per our discussion with @thomasdeleval, we will be allowing user to change their decision only once, so the log note for accept and decline will only be once.
Regarding the second point
Initially, the RFQ allows for acceptance or decline, with a specific acknowledgment message sent for each. Upon confirming the RFQ and issuing a Purchase Order, there are again options to accept or decline, with an activity added for declines. If the user clicks the Decline button in the RFQ email, the PO decline acknowledgment and activity will still be updated.